lighthouse-core/audits/jankless-font.js

```diff
@@ -0,0 +1,97 @@
+  title: 'Fonts with `font-display: optional` are preloaded',
+  /** Title of a Lighthouse audit that provides detail on whether fonts that used `font-display: optional` were preloaded. This descriptive title is shown to users when one or more fonts used `font-display: optional` and were not preloaded. */
+  failureTitle: 'Fonts with `font-display: optional` are not preloaded',
+  /** Description of a Lighthouse audit that tells the user why they should preload fonts if they are using `font-display: optional`. This is displayed after a user expands the section to see more. No character length limits. 'Learn More' becomes link text to additional documentation. */
+  description: 'Preloading fonts that use `font-display: optional` can help reduce layout shifts and improve CLS. [Learn More](https://web.dev/optimize-cls/#web-fonts-causing-foutfoit)',
+};
+
+const str_ = i18n.createMessageInstanceIdFn(__filename, UIStrings);
+
+class JanklessFontAudit extends Audit {
+  /**
+   * @return {LH.Audit.Meta}
+   */
+  static get meta() {
+    return {
+      id: 'jankless-font',
+      title: str_(UIStrings.title),
+      failureTitle: str_(UIStrings.failureTitle),
+      description: str_(UIStrings.description),
+      requiredArtifacts: ['devtoolsLogs', 'URL', 'CSSUsage'],
+    };
+  }
+
+  /**
+   * Finds which font URLs were attempted to be preloaded,
+   * ignoring those that failed to be reused and were requested again.
+   * @param {Array<LH.Artifacts.NetworkRequest>} networkRecords
+   * @return {Set<string>}
+   */
+  static getURLsAttemptedToPreload(networkRecords) {
+    const attemptedURLs = networkRecords
+      .filter(req => req.resourceType === 'Font')
+      .filter(req => req.isLinkPreload)
+      .map(req => req.url);
+
+    return new Set(attemptedURLs);
+  }
+
+  /**
+   * @param {LH.Artifacts} artifacts
+   * @param {LH.Audit.Context} context
+   * @return {Promise<LH.Audit.Product>}
+   */
+  static async audit(artifacts, context) {
+    const devtoolsLog = artifacts.devtoolsLogs[this.DEFAULT_PASS];
+    const networkRecords = await NetworkRecords.request(devtoolsLog, context);
+
+    // Gets the URLs of fonts where font-display: optional.
+    const optionalFontURLs =
+      FontDisplay.findFontDisplayDeclarations(artifacts, PASSING_FONT_DISPLAY_REGEX).passingURLs;
+
+    // Gets the URLs of fonts attempted to be preloaded.
+    const preloadedFontURLs =
+      JanklessFontAudit.getURLsAttemptedToPreload(networkRecords);
+
+    const results = Array.from(optionalFontURLs)
+      .filter(url => !preloadedFontURLs.has(url))
+      .map(url => {
+        return {url: url};
+      });
+
+    /** @type {LH.Audit.Details.Table['headings']} */
+    const headings = [
+      {key: 'url', itemType: 'url', text: str_(i18n.UIStrings.columnURL)},
+      // TODO: show the CLS that could have been saved if font was preloaded.
```

Comment:
this seems similar to our problem of "attribute shifts to the elements that caused it, not the elements that shifted", which we determined to be unfeasible after discussing with Chromies.
@patrickhulce that seem right?

if so, just delete comment :)

Reply:
Yeah it would be awesome if we could but probably not feasible. I'm working on trying to come up with a way to evaluate CLS attribution based on the 3p CWV request-blocking experiments, but even that probably wouldn't help much in this font case (also from recent discussion in #11227 it seems like CLS might not be affected by the preload either way and this is mostly about whether the font is used at all?)

Reply:
I'll delete the comment!